### DISCUSSION TITLE
The Android application developed by react native failed to upload

### DISCUSSION THREAD

**Discussion by BoarDrop** [anisurrahman072/React-Native-Advanced-Guide#9] [Q&A]
The Android application developed by react native failed to upload video files and reported a network error, but other interfaces were normal, and IOS could upload files normally.

**Comment by wadekar9** [comment]
can you share your code snippet?

**Reply by BoarDrop** [reply]
yes, as follow,and i solve my problem , but also thank your concern and answers!

**Comment by anisurrahman072** [comment]
Hi @BoarDrop , great you asked it here 🚀

I handled a similar situation a couple of days ago in a project of my mate. Umm couple of situations could be like his
- Check that you uploaded the video file chunk by chunk (You can search some npm libraries to do that)
- Ensure you used a real device instead of the simulator
- First try to upload a video less than 5 MB then try to upload a bigger one

**Reply by BoarDrop** [reply]
Yes, I checked these problems step by step, and they are not the same. However, thank you very much for your concern and answers.

**Comment by BoarDrop** [comment]
I solved this problem. Just now, I tested other file uploads. First I uploaded pictures, the same code, then I uploaded videos from the Internet, and then I found that they could all be uploaded successfully. Then I became suspicious. It's a problem with my file path. 
My original situation is: I first shoot the video with my mobile phone and store it locally. This is my original code:

 ```
    const formData = new FormData();
    formData.append('key', `${uploadCredentials.data.dir}/${fileName}`);
    formData.append('success_action_status', '200');
    formData.append('policy', uploadCredentials.data.policy);
    formData.append('OSSAccessKeyId', uploadCredentials.data.accessid);
    formData.append('signature', uploadCredentials.data.signature);
    formData.append('callback', uploadCredentials.data.callback);
    formData.append('file', {
      uri: filePath,
      type: fileType,
      name: fileName,
      //data: fileData,
    });
    

    try {
  const response = await axios.post(uploadCredentials.data.host, formData, {
    headers: {'Content-Type': 'multipart/form-data'},
    onUploadProgress: function(progressEvent) {
      if (progressEvent.total) {
        const percentCompleted = Math.round((progressEvent.loaded * 100) / progressEvent.total);
        console.log(`上传进度：${percentCompleted}%`);
        if (percentCompleted == 100) {
          console.log("上传完毕，服务器处理中...");
        }
      } else {
        console.log('上传进度：正在计算...');
      }
    }
  });
  ......
```

Then I found that the url in `formData` needs to be added with the `file://` prefix if it is an Android device, that is:
```
    uri: Platform.OS === 'android' ? `file://${filePath}` : filePath,
```

**Comment by anisurrahman072** [comment]
Great mate 🔥
